### ISSUE TITLE
build dependencies shouldn't be inside Python package

### ISSUE THREAD

**Issue by minrk** [ipython/ipywidgets#287]
package.json, node_modules, etc. have been moved inside the `ipywidgets` directory. These are used to *build* the Javascript, but are not part of the actual result that should be distributed. As a result, these files should be kept *outside* the Python package.

**Comment by SylvainCorlay** [comment]
If I remember correctly, we moved this in order to facilitate the separation of the js into a separate repo, which will have the usual directory structure of an npm package with the package.json at the root...

Unless this changed with the recent modifications in setup.py, it should not be included in the Python package.

**Comment by minrk** [comment]
I guess I don't see how it improves the situation, since the js is still just as mixed with Python as ever. Moving the build files up one level to a normal location shouldn't change that, should it?

**Comment by SylvainCorlay** [comment]
The only benefits were that the internal structure for the javascript is the one that we will have in the separate repo. The separation should not require to many changes in that respect. It also impacted the web[1,2,3] examples.

we could open a PR moving all the js into a top-level `jupyter-js-widgets` directory to effectively complete the separation and have the npm installation of jupyter-js-widgets be part of the pip installation...

**Comment by minrk** [comment]
I think it's okay, I was just surprised and confused. It might be worth taking a stab at making the js a complete package, whose sources are in their own location.

**Comment by ellisonbg** [comment]
I am probably in favor of keeping this stuff top level to make it easy to
prep for the separate JS package.

On Wed, Jan 6, 2016 at 3:08 AM, Min RK <notifications@github.com> wrote:

> I think it's okay, I was just surprised and confused. It might be worth
> taking a stab at making the js a complete package, whose sources are in
> their own location.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ipython/ipywidgets/issues/287#issuecomment-169298856>.
>



-- 
Brian E. Granger
Associate Professor of Physics and Data Science
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com

**Comment by SylvainCorlay** [comment]
The reason why we made it not top level was to prep for the separate JS package :)

**Comment by ellisonbg** [comment]
Ahh, sorry - I wasn't quite following it. I talked to Jon this week about
dev builds not trying to run npm in the setup.py. I think we *should* have
a top level package.json and encourage dev builds to just run the needed
npm commands.

On Wed, Jan 6, 2016 at 6:36 AM, Sylvain Corlay <notifications@github.com>
wrote:

> The reason why we made it not top level was to prep for the separate JS
> package :)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ipython/ipywidgets/issues/287#issuecomment-169338823>.
>



-- 
Brian E. Granger
Associate Professor of Physics and Data Science
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com

**Comment by SylvainCorlay** [comment]
I am not sure I understand, I might be missing something - but we should catch up and chat with @jdfreder (and other interested people) to sync up on this and other small things.

**Comment by jdfreder** [comment]
@ellisonbg Min and I discussed that here: https://github.com/ipython/ipywidgets/pull/278 .  Because of the reasons Min points out there, requiring npm to be executed separately should wait till the separation of repos.  Or, at least the separation of content, like Sylvain suggests.

**Comment by ellisonbg** [comment]
Not sure I follow it all, but I trust you all :)

On Wed, Jan 6, 2016 at 9:09 AM, Jonathan Frederic <notifications@github.com>
wrote:

> @ellisonbg <https://github.com/ellisonbg> Min and I discussed that here:
> #278 <https://github.com/ipython/ipywidgets/pull/278> . Because of the
> reasons Min points out there, requiring npm to be executed separately
> should wait till the separation of repos. Or, at least the separation of
> content, like Sylvain suggests.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ipython/ipywidgets/issues/287#issuecomment-169391637>.
>



-- 
Brian E. Granger
Associate Professor of Physics and Data Science
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com

**Comment by jdfreder** [comment]
@SylvainCorlay we need to address this before we can release.  I'll add a high priority tag to it.

**Comment by SylvainCorlay** [comment]
Yes, we could move all jupyter-js-widgets code in their own directory at the top level.
